src/components/Login.js

```diff
@@ -0,0 +1,222 @@
+
+  const showSingIn = () => {
+    setIsShowSignWhat(SIGN_IN);
+  };
+
+  const showSingUp = () => {
+    setIsShowSignWhat(SIGN_UP);
+  };
+
+  const signUp = () => {
+    createUserWithEmailAndPassword(auth, userEmail, userPassword)
+      .then((userCredential) => {
+        const user = userCredential.user;
+        writeUserData(user.uid);
+        getUserData(user.uid);
+      })
+      .catch((error) => {
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        console.log("errorCode", errorCode);
+        console.log("errorMessage", errorMessage);
+
+        switch (errorCode) {
+          case "auth/wrong-password":
+            alert("密碼錯誤喔");
+            break;
+          case "auth/user-not-found":
+            alert("帳號不存在喔");
+            break;
+          case "auth/weak-password":
+            alert("密碼需至少六碼喔");
+            break;
+          default:
+            alert("登入失敗QQ");
+        }
+      });
+  };
+
+  const signIn = () => {
+    signInWithEmailAndPassword(auth, userEmail, userPassword)
+      .then((userCredential) => {
+        const user = userCredential.user;
+        getUserData(user.uid);
+      })
+      .catch((error) => {
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        console.error("errorCode", errorCode);
+        console.error("errorMessage", errorMessage);
+
+        switch (errorCode) {
+          case "auth/wrong-password":
+            alert("密碼錯誤喔");
+            break;
+          case "auth/user-not-found":
+            alert("帳號不存在喔");
+            break;
+          default:
+            alert("登入失敗QQ");
+        }
+      });
+  };
+
+  const writeUserData = (userId) => {
+    setDoc(doc(db, "user/" + userId), {
+      email: userEmail,
+      password: userPassword,
+      name: userName,
+      link: userLink,
+      uid: userId,
+      follower: [],
+      following: [],
+      pic: userPhoto,
+      role: userRole,
+    });
+    console.log("data written to database");
+  };
+
+  const getUserData = (userId) => {
+    const unsub = onSnapshot(doc(db, "user/" + userId), (doc) => {
+      localStorage.setItem(
+        "userInfo",
+        JSON.stringify({
+          name: doc.data().name,
+          email: doc.data().email,
+          role: doc.data().role,
+          following: doc.data().following,
+          follower: doc.data().follower,
+          pic: doc.data().pic,
+          id: doc.data().uid,
+          link: doc.data().link,
+        })
+      );
+      props.setUserData({
+        name: doc.data().name,
+        email: doc.data().email,
+        role: doc.data().role,
+        following: doc.data().following,
+        follower: doc.data().follower,
+        pic: doc.data().pic,
+        id: doc.data().uid,
+        link: doc.data().link,
+      });
+    });
+  };
+
+  return (
+    <LoginWrapper>
+      <Button onClick={showSingIn}>Sign In</Button>
+      <Button onClick={showSingUp}>Sign Up</Button>
```

Comment:
SignIn and SignUp...
Please use global search to double-check on that

Reply:
modified and checked